src/main/java/at/hannibal2/skyhanni/features/misc/MagicalPowerDisplay.kt

```diff
@@ -0,0 +1,101 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.RenderItemTipEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RegexUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.RegexUtils.matches
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import io.github.moulberry.notenoughupdates.util.stripControlCodes
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+@SkyHanniModule
+object MagicalPowerDisplay {
+    private val config get() = SkyHanniMod.feature.misc.mpDisplay.enabled
+    private val colored get() = SkyHanniMod.feature.misc.mpDisplay.colored
+    private fun isEnabled() = config && LorenzUtils.inSkyBlock
+
+    private val MPMap = mapOf(
+        AccessoryRarity.COMMON to 3, AccessoryRarity.SPECIAL to 3,
+        AccessoryRarity.UNCOMMON to 5, AccessoryRarity.VERY_SPECIAL to 5,
+        AccessoryRarity.RARE to 8,
+        AccessoryRarity.EPIC to 12,
+        AccessoryRarity.LEGENDARY to 16,
+        AccessoryRarity.MYTHIC to 22
+    )
+
+    private val acceptedInvPattern by RepoPattern.pattern(
+        "inv.acceptable",
+        """(Accessory Bag \(\d+/\d+\)|Auctions Browser)"""
+    )
+    private val accessoryLorePattern by RepoPattern.pattern(
+        "accessory.lore",
+        """a?\s*(COMMON|UNCOMMON|RARE|EPIC|LEGENDARY|MYTHIC)\s*(DUNGEON\s*)?ACCESSORY\s*a?"""
```

Comment:
not necessary

Reply:
but then how would I know if it's an accessory?

Reply:
Do we want to show the numbers outside of the accessory bag as well? In that case, we need a check like this, yes.